### PR TITLE
Health simplification 

### LINKED ISSUE #7567: Hunger simulation is unreliable or appears to be
### Environment
- Game Version: 21101020 (latest at the moment of writing)
- OS Version: Win10

### Detailed description of bug

Players report that the hunger simulation is unreliable, going from not needing to eat to having to eat a considerable amount of food to be satisfied. At the very least we can say the hunger system doesn't communicate properly to the players what it is doing under the hood and the reason for the sudden changes.

It was discussed before and we agreed on the hunger system being overcomplicated for no gain in gameplay and being very hard for the players to read.

### Steps to reproduce

1. Play for a while
2. Pay attention to your hunger level

### PR DESCRIPTION
**Pull request by Bod9001**
simplify balancing for heart and lungs,
now the heart can beat as much blood as you need,
the lungs only take as much and produce as much as the lung capacity, 
using ratios to calculate how much should be in the blood and pressure
balances hunger u1 nutriment = one minute Of not being hungry

how long it takes before a species gets hungry is set on the Species data SO

Brings back recursive body parts, but hopefully much simpler

matrix local bounds are cached and Recalculated if tile is placed outside bounds
matrix get at point is 10* faster,
also improves Custom raycast solution

Improved dynamic inventory

~~still need to fix health UI not updating properly~~ ~~critical overlay not updating properly  and communicating damage to the player better~~
~~and checking how damage propagates the body And how to fix player health~~
~~and recoding kidneys~~
~~Adding better indication of hunger~~

### CODE REVIEW

File: UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs [hunk 1]
```diff
@@ -268,20 +272,29 @@ private static MatrixInfo CreateMatrixInfoFromMatrix(Matrix matrix, int id)
...
+				if (Instance.ActiveMatricesList[i].Matrix == Instance.spaceMatrix) continue;
+				if (Instance.ActiveMatricesList[i].Matrix.IsEmptyAt(LocalPos, isServer) == false)
 				{
-					return matrixInfo;
+					return Instance.ActiveMatricesList[i];
 				}
 			}
 
+			SW.Stop();
+			Logger.Log("MatrixInfo AtPoint Took " + SW.ElapsedTicks);
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs]: Stopwatch and log not needed anymore

File: UnityProject/Assets/Scripts/Items/Implants/Organs/Kidneys.cs [hunk 2]
```diff
@@ -26,17 +26,17 @@ public override void SetUpSystems()
...
+			// foreach (var Reagent in WhiteListReagents)
+			// {
+			// 	ContainedGoodReagents[Reagent] = RelatedPart.BloodContainer[Reagent];
+			// }
+			// RelatedPart.BloodContainer.CurrentReagentMix.Clear();
+			//
+			// foreach (var Reagents in ContainedGoodReagents)
+			// {
+			// 	RelatedPart.BloodContainer.CurrentReagentMix.Add(Reagents.Key, Reagents.Value);
+			// }
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/Implants/Organs/Kidneys.cs]: Needed?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Implants/Organs/Kidneys.cs]: " and recoding kidneys" wip

**Comment by corp-0** [comment]
Adding DNM label, let us know when it isn't a wip any more

**Comment by Aranclanos** [comment]
I think it would be best if you opened different PRs with the different changes instead of having matrixmanager changes unrelated to health here and other things for example the nutrient rebalance

**Comment by Bod9001** [comment]
I'll maybe move the matrix stuff out but the nutriment is 
part of the health simplification  
it's in the name health simplification

**Review by ThatDan123 — APPROVED** [review]

File: UnityProject/Assets/Scripts/Items/Implants/Organs/Lungs.cs [hunk 3]
```diff
@@ -48,7 +44,7 @@ public class Lungs : Organ
 	/// The volume of the lung in litres
 	/// </summary>
 	[Tooltip("The volume of the lung in litres")]
-	public float LungSize = 6;
+	public float LungSize = 0.3f;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Items/Implants/Organs/Lungs.cs:47]: That's a very small set of lungs?

Edit: ah, I assume this is at rest.

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**